include/types/access_types.hpp

```diff
@@ -0,0 +1,97 @@
+/***************************************************************************
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename access_patterns.hpp
+ *
+ **************************************************************************/
+
+#ifndef ACCESS_TYPES_HPP
+#define ACCESS_TYPES_HPP
+
+#include "transposition_types.hpp"
+#include <stdexcept>
+
+
+
+/**
+ * @class Access
+ * @brief A wrapper type for Layout, providing common functionality and a safer
+ * interface.
+ */
+class Access {
+  
+ private:
+ /**
+ * @enum Layout
+ * @brief The possible layout options for a matrix, expressed algebraically.
+ */
+enum class Layout { RowMajor, ColMajor };
+  const Layout layout;
+
+  static inline Layout layout_from_transposition(Transposition &t) { 
+     if (t.isNormal()) {
```

Comment:
The problem is that there's also the "conjugate" option - which as I'm aware is subtly different from transposed. In this case, having an "isNormal" is the behaviour that we want most often, so that we know what implementation to pick.